src/java/org/apache/cassandra/schema/SchemaKeyspace.java

```diff
@@ -688,15 +695,58 @@ private static void addColumnToSchemaMutation(TableMetadata table, ColumnMetadat
     {
         AbstractType<?> type = column.type;
         if (type instanceof ReversedType)
-            type = ((ReversedType) type).baseType;
+            type = ((ReversedType<?>) type).baseType;
 
-        builder.update(Columns)
+        Row.SimpleBuilder rowBuilder = builder.update(Columns)
                .row(table.name, column.name.toString())
                .add("column_name_bytes", column.name.bytes)
                .add("kind", column.kind.toString().toLowerCase())
                .add("position", column.position())
                .add("clustering_order", column.clusteringOrder().toString().toLowerCase())
                .add("type", type.asCQL3Type().toString());
+
+        // Dynamic data masking functions shouldn't be attached to columns during rolling upgrades
+        // to avoid sending mutations with columns that are unknown to the old nodes.
+        ColumnMask mask = column.getMask();
+        if (ColumnMask.clusterSupportsMaskedColumns())
+        {
+            if (mask == null)
+            {
+                rowBuilder.delete("mask_keyspace")
```

Comment:
"mask_keyspace" & friends could be constants

Reply:
In other cases yes, but here I would prefer to keep the style of `SchemaKeyspace`, where column names are always referred to with direct strings instead of constants. There are so many of these that probably the class would lose readability if we made all constants.

Reply:
I agree.